prov/efa/src/dgram/efa_ep.c

```diff
@@ -237,11 +237,9 @@ static void efa_ep_destroy(struct efa_ep *ep)
 	efa_ep_destroy_qp(ep->base_ep.qp);
 	fi_freeinfo(ep->base_ep.info);
 	free(ep->base_ep.src_addr);
-	if (ep->util_ep_initialized) {
-		err = ofi_endpoint_close(&ep->base_ep.util_ep);
-		if (err)
-			FI_WARN(&efa_prov, FI_LOG_EP_CTRL, "Unable to close util EP\n");
-	}
+	err = ofi_endpoint_close(&ep->base_ep.util_ep);
```

Comment:
You cannot do this.

calling `ofi_endpoint_close` on an util_ep that has not been initialized will result in segfault

Reply:
Should we add a check on rxr_ep also then? https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/rxr/rxr_ep.c#L795

Reply:
Yes.
or can we make `rxr_ep` use the `util_ep` inside `efa_base_ep`?

Reply:
`rxr_ep` is using the `util_ep` inside `efa_base_ep`. I will move `util_ep_initialized` to `efa_base_ep` and add the check in both endpoints

Reply:
temporarily fixed by adding `util_ep_initialized` check to `rxr_ep`. Will eventually move this to `efa_base_ep_construct`